docs/developer-docs/security/security-best-practices/dos.mdx

```diff
@@ -4,32 +4,72 @@ keywords: [security, concept, denial of service, dos, cycles, captcha]
+- **Monitor cycles usage**: Regularly track your canisters cycles consumption and set alerts for any sudden spikes that may indicate an attack.
+- **Ingress message charging**: While charging for ingress messages (external requests to the canister) is not natively supported, custom solutions could be implemented to make sure that any expensive actions have costs associated with them.
 
-### Security concern
+## 2. Managing Resource Allocation
 
-If an update or query call is expensive in terms of memory used or cycles consumed, this may make it easy for bots to render the canister unusable (e.g. by filling up storage).
+### The "Noisy Neighbor" Problem
 
-### Recommendation
+In a shared resource environment like the Internet Computer, multiple canisters can run on the same subnet. If one canister consumes too many resources (CPU, memory, etc.), it can negatively impact the performance of others on the same subnet. This is known as the "noisy neighbor" problem.
 
-If the dapp offers such operations, consider bot prevention techniques such as adding captchas or proof of work. An example captcha implementation can be found in [Internet Identity](https://github.com/dfinity/internet-identity).
+### Protecting Against Noisy Neighbors
+
+To mitigate the "noisy neighbor" issue, manage your canister's resource allocation effectively:
+- **Memory allocation**: You can set a memory reservation for your canister through the `memory_allocation` setting, ensuring that your canister has enough reserved memory to operate even if other canisters consume large amounts of resources.
+- **Compute reservation**: Similar to memory, you can also reserve compute power for your canister, ensuring it has enough processing power to handle requests. Learn more about managing memory and compute resources in our [Storage](https://internetcomputer.org/docs/current/developer-docs/gas-cost#storage) and [Compute](https://internetcomputer.org/docs/current/developer-docs/gas-cost#compute) guides.
+
+## 3. Managing Cycles
+
+### Cycles and Reserved Cycles
+
+Cycles are a key part of the Internet Computer's resource management system, as they are used to pay for canister operations. To prevent your canister from running out of cycles unexpectedly, you can reserve cycles specifically for future use, helping to ensure consistent operation.
+
+### Managing Reserved Cycles
+
+It's important to monitor and adjust your reserved cycles regularly to meet your canister's needs. Learn more about managing cycles in our [Cycle Bookkeeping and Resource Consumption](https://internetcomputer.org/docs/current/references/ic-interface-spec#cycle-bookkeeping-and-resource-consumption) guide.
```

Comment:
```suggestion
It's important to monitor and adjust your cycles regularly to meet your canister's needs. Learn more about managing cycles in our [Cycle Bookkeeping and Resource Consumption](https://internetcomputer.org/docs/current/references/ic-interface-spec#cycle-bookkeeping-and-resource-consumption) guide.
```
"reserved" cycles might be misunderstood as the cycles set aside from the main balance if the canister makes memory allocations on a subnet that is already short on memory; if we want to go into the details of "reserved" cycles, a separate paragraph might be more suitable